import grpc
import time
from concurrent import futures
import producer_stats_pb2
import producer_stats_pb2_grpc

class ProducerStatsServiceServicer(producer_stats_pb2_grpc.ProducerStatsServiceServicer):
    def SendStats(self, request, context):
        print("\nüìä Received Producer Stats:")
        print(f"‚û°Ô∏è Name: {request.name}")
        print(f"‚û°Ô∏è Client ID: {request.client_id}")
        print(f"‚û°Ô∏è Type: {request.type}")
        print(f"‚û°Ô∏è Age: {request.age}")
        print(f"‚û°Ô∏è Queue Msg Count: {request.msg_cnt}")
        print(f"‚û°Ô∏è TX: {request.tx}")
        print(f"‚û°Ô∏è RX: {request.rx}")
        print(f"‚û°Ô∏è TX Messages: {request.txmsgs}")
        return producer_stats_pb2.Empty()

def serve():
    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
    producer_stats_pb2_grpc.add_ProducerStatsServiceServicer_to_server(ProducerStatsServiceServicer(), server)
    server.add_insecure_port("[::]:50051")
    print("üöÄ gRPC server running on port 50051...")
    server.start()
    server.wait_for_termination()

if __name__ == "__main__":
    serve()
